types/time.go

```diff
@@ -812,7 +813,7 @@ func (d Duration) String() string {
 	}
 
 	fmt.Fprintf(&buf, "%02d:%02d:%02d", hours, minutes, seconds)
-	if d.Fsp > 0 {
+	if d.Fsp > 0 && fraction >= 0 {
```

Comment:
add a comment for this check

Reply:
check fraction seems useless....I removed it